### PR TITLE
refactor(tbs): paralellize aggregate_public_key_shares

### PR DESCRIPTION
**Pull request by dpc**
Re #6398

### CODE REVIEW

**Comment by dpc** [comment]
I need to bench it thorougly. 

But first, @joschisan does the math check out?

File: crypto/tbs/src/lib.rs [hunk 1]
```diff
@@ -188,9 +189,10 @@ pub fn aggregate_public_key_shares(shares: &BTreeMap<u64, PublicKeyShare>) -> Ag
         lagrange_multipliers(shares.keys().cloned().map(Scalar::from).collect())
```
Comment (dpc) [inline crypto/tbs/src/lib.rs:189]: Hmmm... Might need to do something about this one too...

**Comment by dpc** [comment]
I am dpc's dissapointment

```
> env CARGO_PROFILE=release ./scripts/tests/latency-test.sh fm-pay


Baseline (master):

 LATENCY FM PAY: min: 1.0s, avg: 1.1s, median: 1.1s, p90: 1.4s, max: 1.4s, sum: 11.5s



para aggregate_public_key_shares:

 LATENCY FM PAY: min: 0.9s, avg: 1.6s, median: 1.1s, p90: 6.2s, max: 6.2s, sum: 15.8s



para aggregate_public_key_shares + lagrange_multipliers:

 LATENCY FM PAY: min: 1.0s, avg: 1.1s, median: 1.1s, p90: 1.7s, max: 1.7s, sum: 11.2s



para lagrange_multipliers:

 LATENCY FM PAY: min: 1.0s, avg: 1.1s, median: 1.1s, p90: 1.4s, max: 1.4s, sum: 11.2s
```

**Comment by dpc** [comment]
This is probably better done on a higher level (e.g. per note) and maybe already interferes with it.

**Comment by elsirion** [comment]
We barely ever run this fn, afaik it's only required after DKG. I see that we also run it to convert mint server config to mint client config, but that could be easily cached. There's no need to optimize something that runs in a second and shouldn't be called more than once in the first place.

**Comment by dpc** [comment]
> We barely ever run this fn, afaik it's only required after DKG

Hmmm... Possibly I've seen it a lot in the profiling, because we run DKG on start of every test.

**Comment by elsirion** [comment]
If it would massively speed up CI might be worth it, but I don't think we start enough federations sequentially for that.

**Pull request closed without merging**